detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,246 @@
+
+func isEks(utils detectorUtils) (bool, error) {
+	if !isK8s(utils) {
+		return false, nil
+	}
+
+	awsAuth, err := utils.fetchString("GET", k8sSvcURL+authConfigmapPath)
+	if hasProblem(err) {
+		return false, err
+	}
+
+	return awsAuth != "", nil
+}
+
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// eksDetectorUtils is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	fmt.Println("REAL HIT")
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// fetchString is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	authHeader, err := getK8sCredHeader()
+	if hasProblem(err) {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	client := &http.Client{
+		Timeout: millisecondTimeOut * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	response, err := client.Do(request)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	var parsedResp JSONResponse
+	err = json.Unmarshal([]byte(resp), &parsedResp)
+	if hasProblem(err) {
+		return "", err
+	}
+	clusterName := parsedResp.Data.ClusterName
+
+	return clusterName, nil
+}
+
+// getContainerID is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) getContainerID() (string, error) {
+	content, err := readFile(defaultCgroupPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	for _, line := range content {
+		if lineLength := len(line); lineLength > containerIDLength {
+			return line[lineLength-containerIDLength:], nil
+		}
+	}
+
+	return "", nil
+}
+
+func readFile(fileName string) ([]string, error) {
+	file, err := os.Open(fileName)
+	if hasProblem(err) {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+
+	var text []string
+
+	for scanner.Scan() {
+		text = append(text, scanner.Text())
+	}
+
+	return text, nil
+}
+
+func hasProblem(err error) bool {
+	if err == nil {
+		return false
+	}
+	log.Fatalln(err)
```

Comment:
> https://golang.org/pkg/log/#Fatalln

I don't think you want to be exiting the collector for these errors.

Also, while I am not longer a go programmer (so I don't know best practices), I've never seen this `hasProblem(err)` pattern before. Usually you would want custom error log statements that add information in each error case to wrap the error and give more information.